### PR TITLE
distinct map source for selected device

### PR DESCRIPTION
**Pull request by jcardus**
https://github.com/traccar/traccar-web/issues/1156

I tried to use the same source and filter the selected device but it doesn't work. When the source is clustered the feature is never rendered when it's inside a cluster. So I ended up making two sources.

https://github.com/traccar/traccar-web/assets/31902485/5ad3b044-b2cb-477e-a2b1-345eab636202

### CODE REVIEW

File: modern/src/map/MapPositions.js [hunk 1]
```diff
@@ -11,13 +11,14 @@ import { useAttributePreference, usePreference } from '../common/util/preference
 const MapPositions = ({ positions, onClick, showStatus, selectedPosition, titleField }) => {
   const id = useId();
   const clusters = `${id}-clusters`;
-  const direction = `${id}-direction`;
```
Comment (tananaev) [inline modern/src/map/MapPositions.js:14]: Why are we removing this?

File: modern/src/map/MapPositions.js [hunk 2]
```diff
@@ -93,27 +94,53 @@ const MapPositions = ({ positions, onClick, showStatus, selectedPosition, titleF
...
+          'text-font': findFonts(map),
+          'text-size': 12,
+        },
+        paint: {
+          'text-halo-color': 'white',
+          'text-halo-width': 1,
+        },
+      });
+      map.addLayer({
+        id: `direction-${id}`,
```
Comment (jcardus) [inline modern/src/map/MapPositions.js:127]: we need two direction layers, I create the id here

File: modern/src/map/MapPositions.js [hunk 3]
```diff
@@ -162,34 +172,41 @@ const MapPositions = ({ positions, onClick, showStatus, selectedPosition, titleF
...
         map.removeLayer(clusters);
       }
-      if (map.getLayer(direction)) {
-        map.removeLayer(direction);
-      }
-      if (map.getSource(id)) {
-        map.removeSource(id);
-      }
+
+      [id, selected].forEach((id) => {
```
Comment (tananaev) [inline modern/src/map/MapPositions.js:179]: Actually the `id` in the lambda is very confusing. It clashes with the component id.

**Comment by tananaev** [comment]
Merged, thank you.

**Comment by tananaev** [comment]
Looks like this change broke click callbacks for selected devices. Fixed here: https://github.com/traccar/traccar-web/commit/f87566f0bdc7837cf6ac33eadecced1160d019bf.

Also for context:

https://www.traccar.org/forums/topic/route-replay-from-device-popup-window/

**Pull request merged**